Important note to the reader willing to run the code below. After having installed ViZDoom github.com/mwydmuch/ViZDoom/blob/master/doc/Building.md and cloned the following https://github.com/mwydmuch/ViZDoom, execute (for Linux):

> cd ../ViZDoom/examples

> jupyter notebook

then start this notebook...

In this capstone project we want to investigate various Q-Learning techniques to train an agent to play autonomously the game Doom in the ViZDoom environment

We start by discovering the game:
    - we take for granted that vizdoom can be imported (import VizDomm; to install on Linux: >pip3 install vizdoom)
    - what parameters are necessary to load the game
    - what routines are necessary to init and run an episode
    - what inputs the player (agent) can send to the game (the actions)
    - what inputs the player (agent) can receive from the game (the visual inputs & the reward function)
    - no learning takes place here, only random actions are taken from the player (agent): we focus on the interactions

The following code has been adapted from github.com/mwydmuch/ViZDoom/tree/master/examples/python, specifically the file /ViZDoom/examples/python/basic.py, whose credit goes to the developers of the ViZDoom platform, an AI research platform for reinforcement learning from raw visual information: Michal Kempka, Grzegorz Runc, Jakub Toczek & Marek Wydmuch.

# 1. DISCOVERING THE VIZDOOM ENVIRONMENT

In [14]:
from __future__ import print_function
from vizdoom import *

from random import choice
from time import sleep
import numpy as np

### to evaluate benchmarks ###
config_file_path = "../../scenarios/basic.cfg"             # Scenario "Basic"
# config_file_path = "../../scenarios/health_gathering.cfg"    # Scenario "Health gathering supreme"
# config_file_path = "../../scenarios/deadly_corridor.cfg"   # Scenario "Deadly corridor"

def initialize_vizdoom(config_file_path):
    print("Initializing doom...")
    game = DoomGame()
    game.load_config(config_file_path)
    game.set_window_visible(False)
    game.set_mode(Mode.PLAYER)
    game.set_screen_format(ScreenFormat.GRAY8)               # the provided images are grey-scaled
    game.set_screen_resolution(ScreenResolution.RES_640X480)
    game.init()
    print("Doom initialized.")
    return game

game = initialize_vizdoom(config_file_path)


###

'''
# Create DoomGame instance. It will run the game and communicate with the agent.
game = DoomGame()

# CONFIGURATION: we will use game.load_config("../../scenarios/basic.cfg") in the future. 
# For now we check the parameters one by one

# Sets path to the wad file = the game scenario
game.set_doom_scenario_path("../../scenarios/health_gathering_supreme.wad")

# Sets map to start (scenario .wad files can contain many maps).
game.set_doom_map("map01")

# Sets resolution. Default is 320X240
game.set_screen_resolution(ScreenResolution.RES_640X480)

# Sets the screen buffer format
game.set_screen_format(ScreenFormat.RGB24)

# Enables depth buffer
game.set_depth_buffer_enabled(True)

# Enables labeling in game objects labeling
game.set_labels_buffer_enabled(True)

# Enables buffer with top down map of the current episode/level
game.set_automap_buffer_enabled(True)

# Sets other rendering options (all of these options except crosshair are enabled by default)
game.set_render_hud(False)
game.set_render_minimal_hud(False)  # If hud is enabled
game.set_render_crosshair(False)
game.set_render_weapon(True)
game.set_render_decals(False)  # Bullet holes and blood on the walls
game.set_render_particles(False)
game.set_render_effects_sprites(False)  # Smoke and blood
game.set_render_messages(False)  # In-game messages
game.set_render_corpses(False)
game.set_render_screen_flashes(True)  # Effect upon taking damage or picking up items

# Adds buttons that will be allowed for the agent
game.add_available_button(Button.MOVE_LEFT)
game.add_available_button(Button.MOVE_RIGHT)
game.add_available_button(Button.ATTACK)

# Adds game variables that will be included in state
game.add_available_game_variable(GameVariable.AMMO2)

# Causes episodes to finish after 200 tics (actions)
game.set_episode_timeout(200)

# Makes episodes start after 10 tics (~after raising the weapon)
game.set_episode_start_time(10)

# Makes the window appear
game.set_window_visible(True)

# Turns on the sound
game.set_sound_enabled(True)

# Sets the living reward (for each move) to -1
game.set_living_reward(-1)

# Sets ViZDoom mode (PLAYER, ASYNC_PLAYER, SPECTATOR, ASYNC_SPECTATOR, PLAYER mode is default)
game.set_mode(Mode.PLAYER)

# Initializes the game
game.init()

# Define some actions. Each list entry corresponds to declared buttons:
# MOVE_LEFT, MOVE_RIGHT, ATTACK
# 5 more combinations are naturally possible but only 3 are included for visualization
actions = [[True, False, False], [False, True, False], [False, False, True]]
'''

# Run this many episodes
episodes = 50

# we create an array to capture the results and calculate the mean at the end
results = []

# Sets time that will pause the engine after each action (in seconds)
# Without this everything would go too fast
sleep_time = 1.0 / DEFAULT_TICRATE # = 0.028

for i in range(episodes):
    print("Episode #" + str(i + 1))

    # Starts a new episode
    game.new_episode()

    while not game.is_episode_finished():

        # Gets the state
        state = game.get_state()

        # Which consists of:
        n = state.number
        vars = state.game_variables
        screen_buf = state.screen_buffer
        depth_buf = state.depth_buffer
        labels_buf = state.labels_buffer
        automap_buf = state.automap_buffer
        labels = state.labels
        
        # random action and get reward
        r = game.make_action(choice(actions))

        # Prints state's game variables and reward.
        print("State #" + str(n))
        print("Game variables:", vars)
        print("Reward:", r)
        print("=====================")

        if sleep_time > 0:
            sleep(sleep_time)

    # Check the results
    print("Episode finished.")
    print("Total reward:", game.get_total_reward())
    print("************************")
    results.append(game.get_total_reward())

game.close()


# print("results", results)
scores = np.array(results)
print("mean of total reward after 10 runs is:", scores.mean())


Initializing doom...
Doom initialized.
Episode #1
State #1
Game variables: [50.]
Reward: -1.0
State #2
Game variables: [50.]
Reward: -1.0
State #3
Game variables: [50.]
Reward: -1.0
State #4
Game variables: [50.]
Reward: -1.0
State #5
Game variables: [50.]
Reward: -1.0
State #6
Game variables: [50.]
Reward: -1.0
State #7
Game variables: [50.]
Reward: -6.0
State #8
Game variables: [49.]
Reward: -1.0
State #9
Game variables: [49.]
Reward: -1.0
State #10
Game variables: [49.]
Reward: -1.0
State #11
Game variables: [49.]
Reward: -1.0
State #12
Game variables: [49.]
Reward: -1.0
State #13
Game variables: [49.]
Reward: -1.0
State #14
Game variables: [49.]
Reward: -1.0
State #15
Game variables: [49.]
Reward: -1.0
State #16
Game variables: [49.]
Reward: -1.0
State #17
Game variables: [49.]
Reward: -1.0
State #18
Game variables: [49.]
Reward: -1.0
State #19
Game variables: [49.]
Reward: -1.0
State #20
Game variables: [49.]
Reward: -1.0
State #21
Game variables: [49.]
Reward: -1.0
State #22
Game

State #28
Game variables: [48.]
Reward: -1.0
State #29
Game variables: [48.]
Reward: -1.0
State #30
Game variables: [48.]
Reward: -1.0
State #31
Game variables: [48.]
Reward: -1.0
State #32
Game variables: [48.]
Reward: -1.0
State #33
Game variables: [48.]
Reward: -1.0
State #34
Game variables: [48.]
Reward: -1.0
State #35
Game variables: [48.]
Reward: -1.0
State #36
Game variables: [48.]
Reward: -1.0
State #37
Game variables: [48.]
Reward: -1.0
State #38
Game variables: [48.]
Reward: -1.0
State #39
Game variables: [48.]
Reward: -1.0
State #40
Game variables: [48.]
Reward: -1.0
State #41
Game variables: [48.]
Reward: -6.0
State #42
Game variables: [47.]
Reward: -1.0
State #43
Game variables: [47.]
Reward: -1.0
State #44
Game variables: [47.]
Reward: -1.0
State #45
Game variables: [47.]
Reward: -1.0
State #46
Game variables: [47.]
Reward: -1.0
State #47
Game variables: [47.]
Reward: -1.0
State #48
Game variables: [47.]
Reward: -1.0
State #49
Game variables: [47.]
Reward: -1.0
State #50


State #154
Game variables: [41.]
Reward: -1.0
State #155
Game variables: [41.]
Reward: -1.0
State #156
Game variables: [41.]
Reward: -1.0
State #157
Game variables: [41.]
Reward: -1.0
State #158
Game variables: [41.]
Reward: -1.0
State #159
Game variables: [41.]
Reward: -1.0
State #160
Game variables: [41.]
Reward: -6.0
State #161
Game variables: [40.]
Reward: -1.0
State #162
Game variables: [40.]
Reward: -1.0
State #163
Game variables: [40.]
Reward: -1.0
State #164
Game variables: [40.]
Reward: -1.0
State #165
Game variables: [40.]
Reward: -1.0
State #166
Game variables: [40.]
Reward: -1.0
State #167
Game variables: [40.]
Reward: -1.0
State #168
Game variables: [40.]
Reward: -1.0
State #169
Game variables: [40.]
Reward: -1.0
State #170
Game variables: [40.]
Reward: -1.0
State #171
Game variables: [40.]
Reward: -1.0
State #172
Game variables: [40.]
Reward: -1.0
State #173
Game variables: [40.]
Reward: -1.0
State #174
Game variables: [40.]
Reward: -6.0
State #175
Game variables: [39.]
R

State #12
Game variables: [49.]
Reward: -1.0
State #13
Game variables: [49.]
Reward: -1.0
State #14
Game variables: [49.]
Reward: -1.0
State #15
Game variables: [49.]
Reward: -1.0
State #16
Game variables: [49.]
Reward: -1.0
State #17
Game variables: [49.]
Reward: -1.0
State #18
Game variables: [49.]
Reward: -1.0
State #19
Game variables: [49.]
Reward: -6.0
State #20
Game variables: [48.]
Reward: -1.0
State #21
Game variables: [48.]
Reward: -1.0
State #22
Game variables: [48.]
Reward: -1.0
State #23
Game variables: [48.]
Reward: -1.0
State #24
Game variables: [48.]
Reward: -1.0
State #25
Game variables: [48.]
Reward: -1.0
State #26
Game variables: [48.]
Reward: -1.0
State #27
Game variables: [48.]
Reward: -1.0
State #28
Game variables: [48.]
Reward: -1.0
State #29
Game variables: [48.]
Reward: -1.0
State #30
Game variables: [48.]
Reward: -1.0
State #31
Game variables: [48.]
Reward: -1.0
State #32
Game variables: [48.]
Reward: -1.0
State #33
Game variables: [48.]
Reward: -1.0
State #34


State #59
Game variables: [46.]
Reward: -1.0
State #60
Game variables: [46.]
Reward: -1.0
State #61
Game variables: [46.]
Reward: -1.0
State #62
Game variables: [46.]
Reward: -1.0
State #63
Game variables: [46.]
Reward: -1.0
State #64
Game variables: [46.]
Reward: -1.0
State #65
Game variables: [46.]
Reward: -1.0
State #66
Game variables: [46.]
Reward: -1.0
State #67
Game variables: [46.]
Reward: -1.0
State #68
Game variables: [46.]
Reward: -1.0
State #69
Game variables: [46.]
Reward: -1.0
State #70
Game variables: [46.]
Reward: -1.0
State #71
Game variables: [46.]
Reward: -1.0
State #72
Game variables: [46.]
Reward: -1.0
State #73
Game variables: [46.]
Reward: -1.0
State #74
Game variables: [46.]
Reward: -1.0
State #75
Game variables: [46.]
Reward: -1.0
State #76
Game variables: [46.]
Reward: -1.0
State #77
Game variables: [46.]
Reward: -6.0
State #78
Game variables: [45.]
Reward: -1.0
State #79
Game variables: [45.]
Reward: -1.0
State #80
Game variables: [45.]
Reward: -1.0
State #81


State #184
Game variables: [39.]
Reward: -1.0
State #185
Game variables: [39.]
Reward: -1.0
State #186
Game variables: [39.]
Reward: -1.0
State #187
Game variables: [39.]
Reward: -1.0
State #188
Game variables: [39.]
Reward: -1.0
State #189
Game variables: [39.]
Reward: -1.0
State #190
Game variables: [39.]
Reward: -1.0
State #191
Game variables: [39.]
Reward: -1.0
State #192
Game variables: [39.]
Reward: -1.0
State #193
Game variables: [39.]
Reward: -1.0
State #194
Game variables: [39.]
Reward: -1.0
State #195
Game variables: [39.]
Reward: -1.0
State #196
Game variables: [39.]
Reward: -1.0
State #197
Game variables: [39.]
Reward: -1.0
State #198
Game variables: [39.]
Reward: -1.0
State #199
Game variables: [39.]
Reward: -1.0
State #200
Game variables: [39.]
Reward: -1.0
State #201
Game variables: [39.]
Reward: -1.0
State #202
Game variables: [39.]
Reward: -1.0
State #203
Game variables: [39.]
Reward: -6.0
State #204
Game variables: [38.]
Reward: -1.0
State #205
Game variables: [38.]
R

State #8
Game variables: [50.]
Reward: -1.0
State #9
Game variables: [50.]
Reward: -1.0
State #10
Game variables: [50.]
Reward: -1.0
State #11
Game variables: [50.]
Reward: -1.0
State #12
Game variables: [50.]
Reward: -1.0
State #13
Game variables: [50.]
Reward: -1.0
State #14
Game variables: [50.]
Reward: -6.0
State #15
Game variables: [49.]
Reward: -1.0
State #16
Game variables: [49.]
Reward: -1.0
State #17
Game variables: [49.]
Reward: -1.0
State #18
Game variables: [49.]
Reward: -1.0
State #19
Game variables: [49.]
Reward: -1.0
State #20
Game variables: [49.]
Reward: -1.0
State #21
Game variables: [49.]
Reward: -1.0
State #22
Game variables: [49.]
Reward: -1.0
State #23
Game variables: [49.]
Reward: -1.0
State #24
Game variables: [49.]
Reward: -1.0
State #25
Game variables: [49.]
Reward: -1.0
State #26
Game variables: [49.]
Reward: -1.0
State #27
Game variables: [49.]
Reward: -1.0
State #28
Game variables: [49.]
Reward: -6.0
State #29
Game variables: [48.]
Reward: -1.0
State #30
Ga

State #134
Game variables: [43.]
Reward: -1.0
State #135
Game variables: [43.]
Reward: -1.0
State #136
Game variables: [43.]
Reward: -1.0
State #137
Game variables: [43.]
Reward: -1.0
State #138
Game variables: [43.]
Reward: -1.0
State #139
Game variables: [43.]
Reward: -1.0
State #140
Game variables: [43.]
Reward: -1.0
State #141
Game variables: [43.]
Reward: -1.0
State #142
Game variables: [43.]
Reward: -1.0
State #143
Game variables: [43.]
Reward: -1.0
State #144
Game variables: [43.]
Reward: -6.0
State #145
Game variables: [42.]
Reward: -1.0
State #146
Game variables: [42.]
Reward: -1.0
State #147
Game variables: [42.]
Reward: -1.0
State #148
Game variables: [42.]
Reward: -1.0
State #149
Game variables: [42.]
Reward: -1.0
State #150
Game variables: [42.]
Reward: -1.0
State #151
Game variables: [42.]
Reward: -1.0
State #152
Game variables: [42.]
Reward: -1.0
State #153
Game variables: [42.]
Reward: -1.0
State #154
Game variables: [42.]
Reward: -1.0
State #155
Game variables: [42.]
R

State #260
Game variables: [37.]
Reward: -1.0
State #261
Game variables: [37.]
Reward: -6.0
State #262
Game variables: [36.]
Reward: -1.0
State #263
Game variables: [36.]
Reward: -1.0
State #264
Game variables: [36.]
Reward: -1.0
State #265
Game variables: [36.]
Reward: -1.0
State #266
Game variables: [36.]
Reward: -1.0
State #267
Game variables: [36.]
Reward: -1.0
State #268
Game variables: [36.]
Reward: -1.0
State #269
Game variables: [36.]
Reward: -1.0
State #270
Game variables: [36.]
Reward: -1.0
State #271
Game variables: [36.]
Reward: -1.0
State #272
Game variables: [36.]
Reward: -1.0
State #273
Game variables: [36.]
Reward: -1.0
State #274
Game variables: [36.]
Reward: -1.0
State #275
Game variables: [36.]
Reward: -6.0
State #276
Game variables: [35.]
Reward: -1.0
State #277
Game variables: [35.]
Reward: -1.0
State #278
Game variables: [35.]
Reward: -1.0
State #279
Game variables: [35.]
Reward: -1.0
State #280
Game variables: [35.]
Reward: -1.0
State #281
Game variables: [35.]
R

State #82
Game variables: [46.]
Reward: -1.0
State #83
Game variables: [46.]
Reward: -6.0
State #84
Game variables: [45.]
Reward: -1.0
State #85
Game variables: [45.]
Reward: -1.0
State #86
Game variables: [45.]
Reward: -1.0
State #87
Game variables: [45.]
Reward: -1.0
State #88
Game variables: [45.]
Reward: -1.0
State #89
Game variables: [45.]
Reward: -1.0
State #90
Game variables: [45.]
Reward: -1.0
State #91
Game variables: [45.]
Reward: -1.0
State #92
Game variables: [45.]
Reward: -1.0
State #93
Game variables: [45.]
Reward: -1.0
State #94
Game variables: [45.]
Reward: -1.0
State #95
Game variables: [45.]
Reward: -1.0
State #96
Game variables: [45.]
Reward: -1.0
State #97
Game variables: [45.]
Reward: -1.0
State #98
Game variables: [45.]
Reward: -1.0
State #99
Game variables: [45.]
Reward: -1.0
State #100
Game variables: [45.]
Reward: -1.0
State #101
Game variables: [45.]
Reward: -1.0
State #102
Game variables: [45.]
Reward: -6.0
State #103
Game variables: [44.]
Reward: -1.0
State 

State #207
Game variables: [39.]
Reward: -1.0
State #208
Game variables: [39.]
Reward: -1.0
State #209
Game variables: [39.]
Reward: -1.0
State #210
Game variables: [39.]
Reward: -1.0
State #211
Game variables: [39.]
Reward: -1.0
State #212
Game variables: [39.]
Reward: -1.0
State #213
Game variables: [39.]
Reward: -1.0
State #214
Game variables: [39.]
Reward: -1.0
State #215
Game variables: [39.]
Reward: -1.0
State #216
Game variables: [39.]
Reward: -1.0
State #217
Game variables: [39.]
Reward: -1.0
State #218
Game variables: [39.]
Reward: -1.0
State #219
Game variables: [39.]
Reward: -1.0
State #220
Game variables: [39.]
Reward: -1.0
State #221
Game variables: [39.]
Reward: -1.0
State #222
Game variables: [39.]
Reward: -1.0
State #223
Game variables: [39.]
Reward: -6.0
State #224
Game variables: [38.]
Reward: -1.0
State #225
Game variables: [38.]
Reward: -1.0
State #226
Game variables: [38.]
Reward: -1.0
State #227
Game variables: [38.]
Reward: -1.0
State #228
Game variables: [38.]
R

State #27
Game variables: [49.]
Reward: -6.0
State #28
Game variables: [48.]
Reward: -1.0
State #29
Game variables: [48.]
Reward: -1.0
State #30
Game variables: [48.]
Reward: -1.0
State #31
Game variables: [48.]
Reward: -1.0
State #32
Game variables: [48.]
Reward: -1.0
State #33
Game variables: [48.]
Reward: -1.0
State #34
Game variables: [48.]
Reward: -1.0
State #35
Game variables: [48.]
Reward: -1.0
State #36
Game variables: [48.]
Reward: -1.0
State #37
Game variables: [48.]
Reward: -1.0
State #38
Game variables: [48.]
Reward: -1.0
State #39
Game variables: [48.]
Reward: -1.0
State #40
Game variables: [48.]
Reward: -1.0
State #41
Game variables: [48.]
Reward: -1.0
State #42
Game variables: [48.]
Reward: -1.0
State #43
Game variables: [48.]
Reward: -1.0
State #44
Game variables: [48.]
Reward: -1.0
State #45
Game variables: [48.]
Reward: -1.0
State #46
Game variables: [48.]
Reward: -1.0
State #47
Game variables: [48.]
Reward: -1.0
State #48
Game variables: [48.]
Reward: -1.0
State #49


State #40
Game variables: [48.]
Reward: -1.0
State #41
Game variables: [48.]
Reward: -1.0
State #42
Game variables: [48.]
Reward: -1.0
State #43
Game variables: [48.]
Reward: -6.0
State #44
Game variables: [47.]
Reward: -1.0
State #45
Game variables: [47.]
Reward: -1.0
State #46
Game variables: [47.]
Reward: -1.0
State #47
Game variables: [47.]
Reward: -1.0
State #48
Game variables: [47.]
Reward: -1.0
State #49
Game variables: [47.]
Reward: -1.0
State #50
Game variables: [47.]
Reward: -1.0
State #51
Game variables: [47.]
Reward: -1.0
State #52
Game variables: [47.]
Reward: -1.0
State #53
Game variables: [47.]
Reward: -1.0
State #54
Game variables: [47.]
Reward: -1.0
State #55
Game variables: [47.]
Reward: -1.0
State #56
Game variables: [47.]
Reward: -1.0
State #57
Game variables: [47.]
Reward: -6.0
State #58
Game variables: [46.]
Reward: -1.0
State #59
Game variables: [46.]
Reward: -1.0
State #60
Game variables: [46.]
Reward: -1.0
State #61
Game variables: [46.]
Reward: -1.0
State #62


State #165
Game variables: [41.]
Reward: -1.0
State #166
Game variables: [41.]
Reward: -1.0
State #167
Game variables: [41.]
Reward: -1.0
State #168
Game variables: [41.]
Reward: -1.0
State #169
Game variables: [41.]
Reward: -1.0
State #170
Game variables: [41.]
Reward: -1.0
State #171
Game variables: [41.]
Reward: -1.0
State #172
Game variables: [41.]
Reward: -1.0
State #173
Game variables: [41.]
Reward: -1.0
State #174
Game variables: [41.]
Reward: -1.0
State #175
Game variables: [41.]
Reward: -1.0
State #176
Game variables: [41.]
Reward: -6.0
State #177
Game variables: [40.]
Reward: -1.0
State #178
Game variables: [40.]
Reward: -1.0
State #179
Game variables: [40.]
Reward: -1.0
State #180
Game variables: [40.]
Reward: -1.0
State #181
Game variables: [40.]
Reward: -1.0
State #182
Game variables: [40.]
Reward: -1.0
State #183
Game variables: [40.]
Reward: -1.0
State #184
Game variables: [40.]
Reward: -1.0
State #185
Game variables: [40.]
Reward: -1.0
State #186
Game variables: [40.]
R

State #60
Game variables: [47.]
Reward: -1.0
State #61
Game variables: [47.]
Reward: -1.0
State #62
Game variables: [47.]
Reward: -1.0
State #63
Game variables: [47.]
Reward: -1.0
State #64
Game variables: [47.]
Reward: -1.0
State #65
Game variables: [47.]
Reward: -1.0
State #66
Game variables: [47.]
Reward: -1.0
State #67
Game variables: [47.]
Reward: -1.0
State #68
Game variables: [47.]
Reward: -1.0
State #69
Game variables: [47.]
Reward: -1.0
State #70
Game variables: [47.]
Reward: -6.0
State #71
Game variables: [46.]
Reward: -1.0
State #72
Game variables: [46.]
Reward: -1.0
State #73
Game variables: [46.]
Reward: -1.0
State #74
Game variables: [46.]
Reward: -1.0
State #75
Game variables: [46.]
Reward: -1.0
State #76
Game variables: [46.]
Reward: -1.0
State #77
Game variables: [46.]
Reward: -1.0
State #78
Game variables: [46.]
Reward: -1.0
State #79
Game variables: [46.]
Reward: -1.0
State #80
Game variables: [46.]
Reward: -1.0
State #81
Game variables: [46.]
Reward: -1.0
State #82


State #42
Game variables: [48.]
Reward: -1.0
State #43
Game variables: [48.]
Reward: -1.0
State #44
Game variables: [48.]
Reward: -1.0
State #45
Game variables: [48.]
Reward: -1.0
State #46
Game variables: [48.]
Reward: -6.0
State #47
Game variables: [47.]
Reward: -1.0
State #48
Game variables: [47.]
Reward: -1.0
State #49
Game variables: [47.]
Reward: -1.0
State #50
Game variables: [47.]
Reward: -1.0
State #51
Game variables: [47.]
Reward: -1.0
State #52
Game variables: [47.]
Reward: -1.0
State #53
Game variables: [47.]
Reward: -1.0
State #54
Game variables: [47.]
Reward: -1.0
State #55
Game variables: [47.]
Reward: -1.0
State #56
Game variables: [47.]
Reward: -1.0
State #57
Game variables: [47.]
Reward: -1.0
State #58
Game variables: [47.]
Reward: -1.0
State #59
Game variables: [47.]
Reward: -1.0
State #60
Game variables: [47.]
Reward: -1.0
State #61
Game variables: [47.]
Reward: -1.0
State #62
Game variables: [47.]
Reward: -1.0
State #63
Game variables: [47.]
Reward: -1.0
State #64


State #166
Game variables: [41.]
Reward: -1.0
State #167
Game variables: [41.]
Reward: -1.0
State #168
Game variables: [41.]
Reward: -1.0
State #169
Game variables: [41.]
Reward: -1.0
State #170
Game variables: [41.]
Reward: -1.0
State #171
Game variables: [41.]
Reward: -1.0
State #172
Game variables: [41.]
Reward: -1.0
State #173
Game variables: [41.]
Reward: -1.0
State #174
Game variables: [41.]
Reward: -1.0
State #175
Game variables: [41.]
Reward: -1.0
State #176
Game variables: [41.]
Reward: -1.0
State #177
Game variables: [41.]
Reward: -1.0
State #178
Game variables: [41.]
Reward: -1.0
State #179
Game variables: [41.]
Reward: -1.0
State #180
Game variables: [41.]
Reward: -1.0
State #181
Game variables: [41.]
Reward: -1.0
State #182
Game variables: [41.]
Reward: -1.0
State #183
Game variables: [41.]
Reward: -6.0
State #184
Game variables: [40.]
Reward: -1.0
State #185
Game variables: [40.]
Reward: -1.0
State #186
Game variables: [40.]
Reward: -1.0
State #187
Game variables: [40.]
R

State #292
Game variables: [35.]
Reward: -1.0
State #293
Game variables: [35.]
Reward: -1.0
State #294
Game variables: [35.]
Reward: -1.0
State #295
Game variables: [35.]
Reward: -1.0
State #296
Game variables: [35.]
Reward: -1.0
State #297
Game variables: [35.]
Reward: -6.0
State #298
Game variables: [34.]
Reward: -1.0
State #299
Game variables: [34.]
Reward: -1.0
State #300
Game variables: [34.]
Reward: -1.0
Episode finished.
Total reward: -380.0
************************
Episode #12
State #1
Game variables: [50.]
Reward: -1.0
State #2
Game variables: [50.]
Reward: -1.0
State #3
Game variables: [50.]
Reward: -1.0
State #4
Game variables: [50.]
Reward: -1.0
State #5
Game variables: [50.]
Reward: -1.0
State #6
Game variables: [50.]
Reward: -1.0
State #7
Game variables: [50.]
Reward: -6.0
State #8
Game variables: [49.]
Reward: -1.0
State #9
Game variables: [49.]
Reward: -1.0
State #10
Game variables: [49.]
Reward: -1.0
State #11
Game variables: [49.]
Reward: -1.0
State #12
Game variables

State #75
Game variables: [46.]
Reward: -1.0
State #76
Game variables: [46.]
Reward: -1.0
State #77
Game variables: [46.]
Reward: -1.0
State #78
Game variables: [46.]
Reward: -1.0
State #79
Game variables: [46.]
Reward: -1.0
State #80
Game variables: [46.]
Reward: -1.0
State #81
Game variables: [46.]
Reward: -1.0
State #82
Game variables: [46.]
Reward: -1.0
State #83
Game variables: [46.]
Reward: -6.0
State #84
Game variables: [45.]
Reward: -1.0
State #85
Game variables: [45.]
Reward: -1.0
State #86
Game variables: [45.]
Reward: -1.0
State #87
Game variables: [45.]
Reward: -1.0
State #88
Game variables: [45.]
Reward: -1.0
State #89
Game variables: [45.]
Reward: -1.0
State #90
Game variables: [45.]
Reward: -1.0
State #91
Game variables: [45.]
Reward: -1.0
State #92
Game variables: [45.]
Reward: -1.0
State #93
Game variables: [45.]
Reward: -1.0
State #94
Game variables: [45.]
Reward: -1.0
State #95
Game variables: [45.]
Reward: -1.0
State #96
Game variables: [45.]
Reward: -1.0
State #97


State #201
Game variables: [39.]
Reward: -1.0
State #202
Game variables: [39.]
Reward: -1.0
State #203
Game variables: [39.]
Reward: -1.0
State #204
Game variables: [39.]
Reward: -1.0
State #205
Game variables: [39.]
Reward: -1.0
State #206
Game variables: [39.]
Reward: -1.0
State #207
Game variables: [39.]
Reward: -1.0
State #208
Game variables: [39.]
Reward: -1.0
State #209
Game variables: [39.]
Reward: -1.0
State #210
Game variables: [39.]
Reward: -1.0
State #211
Game variables: [39.]
Reward: -1.0
State #212
Game variables: [39.]
Reward: -1.0
State #213
Game variables: [39.]
Reward: -1.0
State #214
Game variables: [39.]
Reward: -1.0
State #215
Game variables: [39.]
Reward: -1.0
State #216
Game variables: [39.]
Reward: -1.0
State #217
Game variables: [39.]
Reward: -1.0
State #218
Game variables: [39.]
Reward: -1.0
State #219
Game variables: [39.]
Reward: -6.0
State #220
Game variables: [38.]
Reward: -1.0
State #221
Game variables: [38.]
Reward: -1.0
State #222
Game variables: [38.]
R

Episode finished.
Total reward: 92.0
************************
Episode #16
State #1
Game variables: [50.]
Reward: -1.0
State #2
Game variables: [50.]
Reward: -1.0
State #3
Game variables: [50.]
Reward: -1.0
State #4
Game variables: [50.]
Reward: -1.0
State #5
Game variables: [50.]
Reward: -6.0
State #6
Game variables: [49.]
Reward: -1.0
State #7
Game variables: [49.]
Reward: -1.0
State #8
Game variables: [49.]
Reward: -1.0
State #9
Game variables: [49.]
Reward: -1.0
State #10
Game variables: [49.]
Reward: -1.0
State #11
Game variables: [49.]
Reward: -1.0
State #12
Game variables: [49.]
Reward: -1.0
State #13
Game variables: [49.]
Reward: -1.0
State #14
Game variables: [49.]
Reward: -1.0
State #15
Game variables: [49.]
Reward: -1.0
State #16
Game variables: [49.]
Reward: -1.0
State #17
Game variables: [49.]
Reward: -1.0
State #18
Game variables: [49.]
Reward: -1.0
State #19
Game variables: [49.]
Reward: -1.0
State #20
Game variables: [49.]
Reward: -1.0
State #21
Game variables: [49.]
Rew

State #122
Game variables: [43.]
Reward: -6.0
State #123
Game variables: [42.]
Reward: -1.0
State #124
Game variables: [42.]
Reward: -1.0
State #125
Game variables: [42.]
Reward: -1.0
State #126
Game variables: [42.]
Reward: -1.0
State #127
Game variables: [42.]
Reward: -1.0
State #128
Game variables: [42.]
Reward: -1.0
State #129
Game variables: [42.]
Reward: -1.0
State #130
Game variables: [42.]
Reward: -1.0
State #131
Game variables: [42.]
Reward: -1.0
State #132
Game variables: [42.]
Reward: -1.0
State #133
Game variables: [42.]
Reward: -1.0
State #134
Game variables: [42.]
Reward: -1.0
State #135
Game variables: [42.]
Reward: -1.0
State #136
Game variables: [42.]
Reward: -6.0
State #137
Game variables: [41.]
Reward: -1.0
State #138
Game variables: [41.]
Reward: -1.0
State #139
Game variables: [41.]
Reward: -1.0
State #140
Game variables: [41.]
Reward: -1.0
State #141
Game variables: [41.]
Reward: -1.0
State #142
Game variables: [41.]
Reward: -1.0
State #143
Game variables: [41.]
R

State #245
Game variables: [36.]
Reward: -1.0
State #246
Game variables: [36.]
Reward: -1.0
State #247
Game variables: [36.]
Reward: -6.0
State #248
Game variables: [35.]
Reward: -1.0
State #249
Game variables: [35.]
Reward: -1.0
State #250
Game variables: [35.]
Reward: -1.0
State #251
Game variables: [35.]
Reward: -1.0
State #252
Game variables: [35.]
Reward: -1.0
State #253
Game variables: [35.]
Reward: -1.0
State #254
Game variables: [35.]
Reward: -1.0
State #255
Game variables: [35.]
Reward: -1.0
State #256
Game variables: [35.]
Reward: -1.0
State #257
Game variables: [35.]
Reward: -1.0
State #258
Game variables: [35.]
Reward: -1.0
State #259
Game variables: [35.]
Reward: -1.0
State #260
Game variables: [35.]
Reward: -1.0
State #261
Game variables: [35.]
Reward: 100.0
State #262
Game variables: [34.]
Reward: -1.0
Episode finished.
Total reward: -236.0
************************
Episode #17
State #1
Game variables: [50.]
Reward: -1.0
State #2
Game variables: [50.]
Reward: -1.0
State #

State #40
Game variables: [48.]
Reward: -1.0
State #41
Game variables: [48.]
Reward: -1.0
State #42
Game variables: [48.]
Reward: -1.0
State #43
Game variables: [48.]
Reward: -1.0
State #44
Game variables: [48.]
Reward: -1.0
State #45
Game variables: [48.]
Reward: -1.0
State #46
Game variables: [48.]
Reward: -1.0
State #47
Game variables: [48.]
Reward: -1.0
State #48
Game variables: [48.]
Reward: -6.0
State #49
Game variables: [47.]
Reward: -1.0
State #50
Game variables: [47.]
Reward: -1.0
State #51
Game variables: [47.]
Reward: -1.0
State #52
Game variables: [47.]
Reward: -1.0
State #53
Game variables: [47.]
Reward: -1.0
State #54
Game variables: [47.]
Reward: -1.0
State #55
Game variables: [47.]
Reward: -1.0
State #56
Game variables: [47.]
Reward: -1.0
State #57
Game variables: [47.]
Reward: -1.0
State #58
Game variables: [47.]
Reward: -1.0
State #59
Game variables: [47.]
Reward: -1.0
State #60
Game variables: [47.]
Reward: -1.0
State #61
Game variables: [47.]
Reward: -1.0
State #62


State #23
Game variables: [49.]
Reward: -1.0
State #24
Game variables: [49.]
Reward: -1.0
State #25
Game variables: [49.]
Reward: -1.0
State #26
Game variables: [49.]
Reward: -1.0
State #27
Game variables: [49.]
Reward: -1.0
State #28
Game variables: [49.]
Reward: -6.0
State #29
Game variables: [48.]
Reward: -1.0
State #30
Game variables: [48.]
Reward: -1.0
State #31
Game variables: [48.]
Reward: -1.0
State #32
Game variables: [48.]
Reward: -1.0
State #33
Game variables: [48.]
Reward: -1.0
State #34
Game variables: [48.]
Reward: -1.0
State #35
Game variables: [48.]
Reward: -1.0
State #36
Game variables: [48.]
Reward: -1.0
State #37
Game variables: [48.]
Reward: -1.0
State #38
Game variables: [48.]
Reward: -1.0
State #39
Game variables: [48.]
Reward: -1.0
State #40
Game variables: [48.]
Reward: -1.0
State #41
Game variables: [48.]
Reward: -1.0
State #42
Game variables: [48.]
Reward: -1.0
State #43
Game variables: [48.]
Reward: -1.0
State #44
Game variables: [48.]
Reward: -1.0
State #45


State #75
Game variables: [46.]
Reward: -1.0
State #76
Game variables: [46.]
Reward: -1.0
State #77
Game variables: [46.]
Reward: -1.0
State #78
Game variables: [46.]
Reward: -1.0
State #79
Game variables: [46.]
Reward: -1.0
State #80
Game variables: [46.]
Reward: -1.0
State #81
Game variables: [46.]
Reward: -1.0
State #82
Game variables: [46.]
Reward: -1.0
State #83
Game variables: [46.]
Reward: -1.0
State #84
Game variables: [46.]
Reward: -1.0
State #85
Game variables: [46.]
Reward: -6.0
State #86
Game variables: [45.]
Reward: -1.0
State #87
Game variables: [45.]
Reward: -1.0
State #88
Game variables: [45.]
Reward: -1.0
State #89
Game variables: [45.]
Reward: -1.0
State #90
Game variables: [45.]
Reward: -1.0
State #91
Game variables: [45.]
Reward: -1.0
State #92
Game variables: [45.]
Reward: -1.0
State #93
Game variables: [45.]
Reward: -1.0
State #94
Game variables: [45.]
Reward: -1.0
State #95
Game variables: [45.]
Reward: -1.0
State #96
Game variables: [45.]
Reward: -1.0
State #97


State #200
Game variables: [39.]
Reward: -1.0
State #201
Game variables: [39.]
Reward: -1.0
State #202
Game variables: [39.]
Reward: -1.0
State #203
Game variables: [39.]
Reward: -1.0
State #204
Game variables: [39.]
Reward: -1.0
State #205
Game variables: [39.]
Reward: -1.0
State #206
Game variables: [39.]
Reward: -1.0
State #207
Game variables: [39.]
Reward: -1.0
State #208
Game variables: [39.]
Reward: -1.0
State #209
Game variables: [39.]
Reward: -1.0
State #210
Game variables: [39.]
Reward: -1.0
State #211
Game variables: [39.]
Reward: -1.0
State #212
Game variables: [39.]
Reward: -1.0
State #213
Game variables: [39.]
Reward: -1.0
State #214
Game variables: [39.]
Reward: -1.0
State #215
Game variables: [39.]
Reward: -1.0
State #216
Game variables: [39.]
Reward: -1.0
State #217
Game variables: [39.]
Reward: -1.0
State #218
Game variables: [39.]
Reward: -1.0
State #219
Game variables: [39.]
Reward: -1.0
State #220
Game variables: [39.]
Reward: -1.0
State #221
Game variables: [39.]
R

State #24
Game variables: [48.]
Reward: -1.0
State #25
Game variables: [48.]
Reward: -1.0
State #26
Game variables: [48.]
Reward: -1.0
State #27
Game variables: [48.]
Reward: -1.0
State #28
Game variables: [48.]
Reward: -1.0
State #29
Game variables: [48.]
Reward: -1.0
State #30
Game variables: [48.]
Reward: -1.0
State #31
Game variables: [48.]
Reward: -1.0
State #32
Game variables: [48.]
Reward: -1.0
State #33
Game variables: [48.]
Reward: -1.0
State #34
Game variables: [48.]
Reward: -1.0
State #35
Game variables: [48.]
Reward: -6.0
State #36
Game variables: [47.]
Reward: -1.0
State #37
Game variables: [47.]
Reward: -1.0
State #38
Game variables: [47.]
Reward: -1.0
State #39
Game variables: [47.]
Reward: -1.0
State #40
Game variables: [47.]
Reward: -1.0
State #41
Game variables: [47.]
Reward: -1.0
State #42
Game variables: [47.]
Reward: -1.0
State #43
Game variables: [47.]
Reward: -1.0
State #44
Game variables: [47.]
Reward: -1.0
State #45
Game variables: [47.]
Reward: -1.0
State #46


State #149
Game variables: [42.]
Reward: -1.0
State #150
Game variables: [42.]
Reward: -1.0
State #151
Game variables: [42.]
Reward: -1.0
State #152
Game variables: [42.]
Reward: -6.0
State #153
Game variables: [41.]
Reward: -1.0
State #154
Game variables: [41.]
Reward: -1.0
State #155
Game variables: [41.]
Reward: -1.0
State #156
Game variables: [41.]
Reward: -1.0
State #157
Game variables: [41.]
Reward: -1.0
State #158
Game variables: [41.]
Reward: -1.0
State #159
Game variables: [41.]
Reward: -1.0
State #160
Game variables: [41.]
Reward: -1.0
State #161
Game variables: [41.]
Reward: -1.0
State #162
Game variables: [41.]
Reward: -1.0
State #163
Game variables: [41.]
Reward: -1.0
State #164
Game variables: [41.]
Reward: -1.0
State #165
Game variables: [41.]
Reward: -1.0
State #166
Game variables: [41.]
Reward: -6.0
State #167
Game variables: [40.]
Reward: -1.0
State #168
Game variables: [40.]
Reward: -1.0
State #169
Game variables: [40.]
Reward: -1.0
State #170
Game variables: [40.]
R

State #274
Game variables: [35.]
Reward: -6.0
State #275
Game variables: [34.]
Reward: -1.0
State #276
Game variables: [34.]
Reward: -1.0
State #277
Game variables: [34.]
Reward: -1.0
State #278
Game variables: [34.]
Reward: -1.0
State #279
Game variables: [34.]
Reward: -1.0
State #280
Game variables: [34.]
Reward: -1.0
State #281
Game variables: [34.]
Reward: -1.0
State #282
Game variables: [34.]
Reward: -1.0
State #283
Game variables: [34.]
Reward: -1.0
State #284
Game variables: [34.]
Reward: -1.0
State #285
Game variables: [34.]
Reward: -1.0
State #286
Game variables: [34.]
Reward: -1.0
State #287
Game variables: [34.]
Reward: -1.0
State #288
Game variables: [34.]
Reward: -6.0
State #289
Game variables: [33.]
Reward: -1.0
State #290
Game variables: [33.]
Reward: -1.0
State #291
Game variables: [33.]
Reward: -1.0
State #292
Game variables: [33.]
Reward: -1.0
State #293
Game variables: [33.]
Reward: -1.0
State #294
Game variables: [33.]
Reward: -1.0
State #295
Game variables: [33.]
R

State #97
Game variables: [44.]
Reward: -1.0
State #98
Game variables: [44.]
Reward: -1.0
State #99
Game variables: [44.]
Reward: -1.0
State #100
Game variables: [44.]
Reward: -1.0
State #101
Game variables: [44.]
Reward: -1.0
State #102
Game variables: [44.]
Reward: -1.0
State #103
Game variables: [44.]
Reward: -1.0
State #104
Game variables: [44.]
Reward: -1.0
State #105
Game variables: [44.]
Reward: -1.0
State #106
Game variables: [44.]
Reward: -1.0
State #107
Game variables: [44.]
Reward: -1.0
State #108
Game variables: [44.]
Reward: -6.0
State #109
Game variables: [43.]
Reward: -1.0
State #110
Game variables: [43.]
Reward: -1.0
State #111
Game variables: [43.]
Reward: -1.0
State #112
Game variables: [43.]
Reward: -1.0
State #113
Game variables: [43.]
Reward: -1.0
State #114
Game variables: [43.]
Reward: -1.0
State #115
Game variables: [43.]
Reward: -1.0
State #116
Game variables: [43.]
Reward: -1.0
State #117
Game variables: [43.]
Reward: -1.0
State #118
Game variables: [43.]
Rewa

State #223
Game variables: [38.]
Reward: -1.0
State #224
Game variables: [38.]
Reward: -1.0
State #225
Game variables: [38.]
Reward: -1.0
State #226
Game variables: [38.]
Reward: -6.0
State #227
Game variables: [37.]
Reward: -1.0
State #228
Game variables: [37.]
Reward: -1.0
State #229
Game variables: [37.]
Reward: -1.0
State #230
Game variables: [37.]
Reward: -1.0
State #231
Game variables: [37.]
Reward: -1.0
State #232
Game variables: [37.]
Reward: -1.0
State #233
Game variables: [37.]
Reward: -1.0
State #234
Game variables: [37.]
Reward: -1.0
State #235
Game variables: [37.]
Reward: -1.0
State #236
Game variables: [37.]
Reward: -1.0
State #237
Game variables: [37.]
Reward: -1.0
State #238
Game variables: [37.]
Reward: -1.0
State #239
Game variables: [37.]
Reward: -1.0
State #240
Game variables: [37.]
Reward: -1.0
State #241
Game variables: [37.]
Reward: -1.0
State #242
Game variables: [37.]
Reward: -1.0
State #243
Game variables: [37.]
Reward: -1.0
State #244
Game variables: [37.]
R

State #42
Game variables: [48.]
Reward: -1.0
State #43
Game variables: [48.]
Reward: -1.0
State #44
Game variables: [48.]
Reward: -1.0
State #45
Game variables: [48.]
Reward: -1.0
State #46
Game variables: [48.]
Reward: -1.0
State #47
Game variables: [48.]
Reward: -1.0
State #48
Game variables: [48.]
Reward: -1.0
State #49
Game variables: [48.]
Reward: -1.0
State #50
Game variables: [48.]
Reward: -1.0
State #51
Game variables: [48.]
Reward: -1.0
State #52
Game variables: [48.]
Reward: -6.0
State #53
Game variables: [47.]
Reward: -1.0
State #54
Game variables: [47.]
Reward: -1.0
State #55
Game variables: [47.]
Reward: -1.0
State #56
Game variables: [47.]
Reward: -1.0
State #57
Game variables: [47.]
Reward: -1.0
State #58
Game variables: [47.]
Reward: -1.0
State #59
Game variables: [47.]
Reward: -1.0
State #60
Game variables: [47.]
Reward: -1.0
State #61
Game variables: [47.]
Reward: -1.0
State #62
Game variables: [47.]
Reward: -1.0
State #63
Game variables: [47.]
Reward: -1.0
State #64


State #80
Game variables: [46.]
Reward: -1.0
State #81
Game variables: [46.]
Reward: -1.0
State #82
Game variables: [46.]
Reward: -1.0
State #83
Game variables: [46.]
Reward: -1.0
State #84
Game variables: [46.]
Reward: -1.0
State #85
Game variables: [46.]
Reward: -1.0
State #86
Game variables: [46.]
Reward: -6.0
State #87
Game variables: [45.]
Reward: -1.0
State #88
Game variables: [45.]
Reward: -1.0
State #89
Game variables: [45.]
Reward: -1.0
State #90
Game variables: [45.]
Reward: -1.0
State #91
Game variables: [45.]
Reward: -1.0
State #92
Game variables: [45.]
Reward: -1.0
State #93
Game variables: [45.]
Reward: -1.0
State #94
Game variables: [45.]
Reward: -1.0
State #95
Game variables: [45.]
Reward: -1.0
State #96
Game variables: [45.]
Reward: -1.0
State #97
Game variables: [45.]
Reward: -1.0
State #98
Game variables: [45.]
Reward: -1.0
State #99
Game variables: [45.]
Reward: -1.0
State #100
Game variables: [45.]
Reward: -1.0
State #101
Game variables: [45.]
Reward: -1.0
State #1

State #206
Game variables: [39.]
Reward: -1.0
State #207
Game variables: [39.]
Reward: -1.0
State #208
Game variables: [39.]
Reward: -1.0
State #209
Game variables: [39.]
Reward: -1.0
State #210
Game variables: [39.]
Reward: -1.0
State #211
Game variables: [39.]
Reward: -1.0
State #212
Game variables: [39.]
Reward: -1.0
State #213
Game variables: [39.]
Reward: -1.0
State #214
Game variables: [39.]
Reward: -1.0
State #215
Game variables: [39.]
Reward: -1.0
State #216
Game variables: [39.]
Reward: -1.0
State #217
Game variables: [39.]
Reward: -1.0
State #218
Game variables: [39.]
Reward: 100.0
State #219
Game variables: [38.]
Reward: -1.0
Episode finished.
Total reward: -173.0
************************
Episode #27
State #1
Game variables: [50.]
Reward: -1.0
State #2
Game variables: [50.]
Reward: -1.0
State #3
Game variables: [50.]
Reward: -1.0
State #4
Game variables: [50.]
Reward: -1.0
State #5
Game variables: [50.]
Reward: -6.0
State #6
Game variables: [49.]
Reward: -1.0
State #7
Game v

State #110
Game variables: [44.]
Reward: -1.0
State #111
Game variables: [44.]
Reward: -1.0
State #112
Game variables: [44.]
Reward: -1.0
State #113
Game variables: [44.]
Reward: -1.0
State #114
Game variables: [44.]
Reward: -1.0
State #115
Game variables: [44.]
Reward: -1.0
State #116
Game variables: [44.]
Reward: -1.0
State #117
Game variables: [44.]
Reward: -6.0
State #118
Game variables: [43.]
Reward: -1.0
State #119
Game variables: [43.]
Reward: -1.0
State #120
Game variables: [43.]
Reward: -1.0
State #121
Game variables: [43.]
Reward: -1.0
State #122
Game variables: [43.]
Reward: -1.0
State #123
Game variables: [43.]
Reward: -1.0
State #124
Game variables: [43.]
Reward: -1.0
State #125
Game variables: [43.]
Reward: -1.0
State #126
Game variables: [43.]
Reward: -1.0
State #127
Game variables: [43.]
Reward: -1.0
State #128
Game variables: [43.]
Reward: -1.0
State #129
Game variables: [43.]
Reward: -1.0
State #130
Game variables: [43.]
Reward: -1.0
State #131
Game variables: [43.]
R

State #236
Game variables: [37.]
Reward: -1.0
State #237
Game variables: [37.]
Reward: -1.0
State #238
Game variables: [37.]
Reward: -1.0
State #239
Game variables: [37.]
Reward: -1.0
State #240
Game variables: [37.]
Reward: -1.0
State #241
Game variables: [37.]
Reward: -1.0
State #242
Game variables: [37.]
Reward: -1.0
State #243
Game variables: [37.]
Reward: -1.0
State #244
Game variables: [37.]
Reward: -1.0
State #245
Game variables: [37.]
Reward: -1.0
State #246
Game variables: [37.]
Reward: -1.0
State #247
Game variables: [37.]
Reward: -1.0
State #248
Game variables: [37.]
Reward: -1.0
State #249
Game variables: [37.]
Reward: -1.0
State #250
Game variables: [37.]
Reward: -1.0
State #251
Game variables: [37.]
Reward: -1.0
State #252
Game variables: [37.]
Reward: -1.0
State #253
Game variables: [37.]
Reward: -1.0
State #254
Game variables: [37.]
Reward: -1.0
State #255
Game variables: [37.]
Reward: -1.0
State #256
Game variables: [37.]
Reward: -1.0
State #257
Game variables: [37.]
R

State #58
Game variables: [47.]
Reward: -1.0
State #59
Game variables: [47.]
Reward: -1.0
State #60
Game variables: [47.]
Reward: -1.0
State #61
Game variables: [47.]
Reward: -1.0
State #62
Game variables: [47.]
Reward: -1.0
State #63
Game variables: [47.]
Reward: -1.0
State #64
Game variables: [47.]
Reward: -1.0
State #65
Game variables: [47.]
Reward: -1.0
State #66
Game variables: [47.]
Reward: -1.0
State #67
Game variables: [47.]
Reward: -1.0
State #68
Game variables: [47.]
Reward: -1.0
State #69
Game variables: [47.]
Reward: -6.0
State #70
Game variables: [46.]
Reward: -1.0
State #71
Game variables: [46.]
Reward: -1.0
State #72
Game variables: [46.]
Reward: -1.0
State #73
Game variables: [46.]
Reward: -1.0
State #74
Game variables: [46.]
Reward: -1.0
State #75
Game variables: [46.]
Reward: -1.0
State #76
Game variables: [46.]
Reward: -1.0
State #77
Game variables: [46.]
Reward: -1.0
State #78
Game variables: [46.]
Reward: -1.0
State #79
Game variables: [46.]
Reward: -1.0
State #80


State #34
Game variables: [48.]
Reward: -1.0
State #35
Game variables: [48.]
Reward: -1.0
State #36
Game variables: [48.]
Reward: -1.0
State #37
Game variables: [48.]
Reward: -1.0
State #38
Game variables: [48.]
Reward: -1.0
State #39
Game variables: [48.]
Reward: -1.0
State #40
Game variables: [48.]
Reward: -1.0
State #41
Game variables: [48.]
Reward: -1.0
State #42
Game variables: [48.]
Reward: -1.0
State #43
Game variables: [48.]
Reward: -1.0
State #44
Game variables: [48.]
Reward: -1.0
State #45
Game variables: [48.]
Reward: -1.0
State #46
Game variables: [48.]
Reward: -1.0
State #47
Game variables: [48.]
Reward: -1.0
State #48
Game variables: [48.]
Reward: -1.0
State #49
Game variables: [48.]
Reward: -1.0
State #50
Game variables: [48.]
Reward: -1.0
State #51
Game variables: [48.]
Reward: -1.0
State #52
Game variables: [48.]
Reward: -1.0
State #53
Game variables: [48.]
Reward: -1.0
State #54
Game variables: [48.]
Reward: -1.0
State #55
Game variables: [48.]
Reward: -6.0
State #56


State #158
Game variables: [42.]
Reward: -1.0
State #159
Game variables: [42.]
Reward: -1.0
State #160
Game variables: [42.]
Reward: -1.0
State #161
Game variables: [42.]
Reward: -1.0
State #162
Game variables: [42.]
Reward: -1.0
State #163
Game variables: [42.]
Reward: -1.0
State #164
Game variables: [42.]
Reward: -1.0
State #165
Game variables: [42.]
Reward: -1.0
State #166
Game variables: [42.]
Reward: -1.0
State #167
Game variables: [42.]
Reward: -1.0
State #168
Game variables: [42.]
Reward: -1.0
State #169
Game variables: [42.]
Reward: -6.0
State #170
Game variables: [41.]
Reward: -1.0
State #171
Game variables: [41.]
Reward: -1.0
State #172
Game variables: [41.]
Reward: -1.0
State #173
Game variables: [41.]
Reward: -1.0
State #174
Game variables: [41.]
Reward: -1.0
State #175
Game variables: [41.]
Reward: -1.0
State #176
Game variables: [41.]
Reward: -1.0
State #177
Game variables: [41.]
Reward: -1.0
State #178
Game variables: [41.]
Reward: -1.0
State #179
Game variables: [41.]
R

State #44
Game variables: [48.]
Reward: -1.0
State #45
Game variables: [48.]
Reward: -1.0
State #46
Game variables: [48.]
Reward: -1.0
State #47
Game variables: [48.]
Reward: -1.0
State #48
Game variables: [48.]
Reward: -6.0
State #49
Game variables: [47.]
Reward: -1.0
State #50
Game variables: [47.]
Reward: -1.0
State #51
Game variables: [47.]
Reward: -1.0
State #52
Game variables: [47.]
Reward: -1.0
State #53
Game variables: [47.]
Reward: -1.0
State #54
Game variables: [47.]
Reward: -1.0
State #55
Game variables: [47.]
Reward: -1.0
State #56
Game variables: [47.]
Reward: -1.0
State #57
Game variables: [47.]
Reward: -1.0
State #58
Game variables: [47.]
Reward: -1.0
State #59
Game variables: [47.]
Reward: -1.0
State #60
Game variables: [47.]
Reward: -1.0
State #61
Game variables: [47.]
Reward: -1.0
State #62
Game variables: [47.]
Reward: -1.0
State #63
Game variables: [47.]
Reward: -1.0
State #64
Game variables: [47.]
Reward: -1.0
State #65
Game variables: [47.]
Reward: -1.0
State #66


State #171
Game variables: [40.]
Reward: -1.0
State #172
Game variables: [40.]
Reward: -1.0
State #173
Game variables: [40.]
Reward: -1.0
State #174
Game variables: [40.]
Reward: -1.0
State #175
Game variables: [40.]
Reward: -1.0
State #176
Game variables: [40.]
Reward: -1.0
State #177
Game variables: [40.]
Reward: -1.0
State #178
Game variables: [40.]
Reward: -1.0
State #179
Game variables: [40.]
Reward: -1.0
State #180
Game variables: [40.]
Reward: -1.0
State #181
Game variables: [40.]
Reward: -1.0
State #182
Game variables: [40.]
Reward: -1.0
State #183
Game variables: [40.]
Reward: -1.0
State #184
Game variables: [40.]
Reward: -6.0
State #185
Game variables: [39.]
Reward: -1.0
State #186
Game variables: [39.]
Reward: -1.0
State #187
Game variables: [39.]
Reward: -1.0
State #188
Game variables: [39.]
Reward: -1.0
State #189
Game variables: [39.]
Reward: -1.0
State #190
Game variables: [39.]
Reward: -1.0
State #191
Game variables: [39.]
Reward: -1.0
State #192
Game variables: [39.]
R

State #296
Game variables: [32.]
Reward: -1.0
State #297
Game variables: [32.]
Reward: -1.0
State #298
Game variables: [32.]
Reward: -1.0
State #299
Game variables: [32.]
Reward: -1.0
State #300
Game variables: [32.]
Reward: -1.0
Episode finished.
Total reward: -390.0
************************
Episode #32
State #1
Game variables: [50.]
Reward: -1.0
State #2
Game variables: [50.]
Reward: -1.0
State #3
Game variables: [50.]
Reward: -1.0
State #4
Game variables: [50.]
Reward: -1.0
State #5
Game variables: [50.]
Reward: -6.0
State #6
Game variables: [49.]
Reward: -1.0
State #7
Game variables: [49.]
Reward: -1.0
State #8
Game variables: [49.]
Reward: -1.0
State #9
Game variables: [49.]
Reward: -1.0
State #10
Game variables: [49.]
Reward: -1.0
State #11
Game variables: [49.]
Reward: -1.0
State #12
Game variables: [49.]
Reward: -1.0
State #13
Game variables: [49.]
Reward: -1.0
State #14
Game variables: [49.]
Reward: -1.0
State #15
Game variables: [49.]
Reward: -1.0
State #16
Game variables: [4

State #120
Game variables: [44.]
Reward: -1.0
State #121
Game variables: [44.]
Reward: -1.0
State #122
Game variables: [44.]
Reward: -1.0
State #123
Game variables: [44.]
Reward: -1.0
State #124
Game variables: [44.]
Reward: -1.0
State #125
Game variables: [44.]
Reward: -6.0
State #126
Game variables: [43.]
Reward: -1.0
State #127
Game variables: [43.]
Reward: -1.0
State #128
Game variables: [43.]
Reward: -1.0
State #129
Game variables: [43.]
Reward: -1.0
State #130
Game variables: [43.]
Reward: -1.0
State #131
Game variables: [43.]
Reward: -1.0
State #132
Game variables: [43.]
Reward: -1.0
State #133
Game variables: [43.]
Reward: -1.0
State #134
Game variables: [43.]
Reward: -1.0
State #135
Game variables: [43.]
Reward: -1.0
State #136
Game variables: [43.]
Reward: -1.0
State #137
Game variables: [43.]
Reward: -1.0
State #138
Game variables: [43.]
Reward: -1.0
State #139
Game variables: [43.]
Reward: -1.0
State #140
Game variables: [43.]
Reward: -1.0
State #141
Game variables: [43.]
R

State #245
Game variables: [37.]
Reward: -1.0
State #246
Game variables: [37.]
Reward: -1.0
State #247
Game variables: [37.]
Reward: -1.0
State #248
Game variables: [37.]
Reward: -1.0
State #249
Game variables: [37.]
Reward: -1.0
State #250
Game variables: [37.]
Reward: -6.0
State #251
Game variables: [36.]
Reward: -1.0
State #252
Game variables: [36.]
Reward: -1.0
State #253
Game variables: [36.]
Reward: -1.0
State #254
Game variables: [36.]
Reward: -1.0
State #255
Game variables: [36.]
Reward: -1.0
State #256
Game variables: [36.]
Reward: -1.0
State #257
Game variables: [36.]
Reward: -1.0
State #258
Game variables: [36.]
Reward: -1.0
State #259
Game variables: [36.]
Reward: -1.0
State #260
Game variables: [36.]
Reward: -1.0
State #261
Game variables: [36.]
Reward: -1.0
State #262
Game variables: [36.]
Reward: -1.0
State #263
Game variables: [36.]
Reward: -1.0
State #264
Game variables: [36.]
Reward: -6.0
State #265
Game variables: [35.]
Reward: -1.0
State #266
Game variables: [35.]
R

State #71
Game variables: [46.]
Reward: -1.0
State #72
Game variables: [46.]
Reward: -1.0
State #73
Game variables: [46.]
Reward: -1.0
State #74
Game variables: [46.]
Reward: -1.0
State #75
Game variables: [46.]
Reward: -1.0
State #76
Game variables: [46.]
Reward: -1.0
State #77
Game variables: [46.]
Reward: -1.0
State #78
Game variables: [46.]
Reward: -1.0
State #79
Game variables: [46.]
Reward: -1.0
State #80
Game variables: [46.]
Reward: -1.0
State #81
Game variables: [46.]
Reward: -1.0
State #82
Game variables: [46.]
Reward: -1.0
State #83
Game variables: [46.]
Reward: -1.0
State #84
Game variables: [46.]
Reward: -1.0
State #85
Game variables: [46.]
Reward: -6.0
State #86
Game variables: [45.]
Reward: -1.0
State #87
Game variables: [45.]
Reward: -1.0
State #88
Game variables: [45.]
Reward: -1.0
State #89
Game variables: [45.]
Reward: -1.0
State #90
Game variables: [45.]
Reward: -1.0
State #91
Game variables: [45.]
Reward: -1.0
State #92
Game variables: [45.]
Reward: -1.0
State #93


State #196
Game variables: [40.]
Reward: -1.0
State #197
Game variables: [40.]
Reward: -1.0
State #198
Game variables: [40.]
Reward: -6.0
State #199
Game variables: [39.]
Reward: -1.0
State #200
Game variables: [39.]
Reward: -1.0
State #201
Game variables: [39.]
Reward: -1.0
State #202
Game variables: [39.]
Reward: -1.0
State #203
Game variables: [39.]
Reward: -1.0
State #204
Game variables: [39.]
Reward: -1.0
State #205
Game variables: [39.]
Reward: -1.0
State #206
Game variables: [39.]
Reward: -1.0
State #207
Game variables: [39.]
Reward: -1.0
State #208
Game variables: [39.]
Reward: -1.0
State #209
Game variables: [39.]
Reward: -1.0
State #210
Game variables: [39.]
Reward: -1.0
State #211
Game variables: [39.]
Reward: -1.0
State #212
Game variables: [39.]
Reward: -1.0
State #213
Game variables: [39.]
Reward: -1.0
State #214
Game variables: [39.]
Reward: -1.0
State #215
Game variables: [39.]
Reward: -1.0
State #216
Game variables: [39.]
Reward: -1.0
State #217
Game variables: [39.]
R

State #9
Game variables: [49.]
Reward: -1.0
State #10
Game variables: [49.]
Reward: -1.0
State #11
Game variables: [49.]
Reward: -1.0
State #12
Game variables: [49.]
Reward: -1.0
State #13
Game variables: [49.]
Reward: -1.0
State #14
Game variables: [49.]
Reward: -1.0
State #15
Game variables: [49.]
Reward: -1.0
State #16
Game variables: [49.]
Reward: -1.0
State #17
Game variables: [49.]
Reward: -1.0
State #18
Game variables: [49.]
Reward: -1.0
State #19
Game variables: [49.]
Reward: -6.0
State #20
Game variables: [48.]
Reward: -1.0
State #21
Game variables: [48.]
Reward: -1.0
State #22
Game variables: [48.]
Reward: -1.0
State #23
Game variables: [48.]
Reward: -1.0
State #24
Game variables: [48.]
Reward: -1.0
State #25
Game variables: [48.]
Reward: -1.0
State #26
Game variables: [48.]
Reward: -1.0
State #27
Game variables: [48.]
Reward: -1.0
State #28
Game variables: [48.]
Reward: -1.0
State #29
Game variables: [48.]
Reward: -1.0
State #30
Game variables: [48.]
Reward: -1.0
State #31
G

State #20
Game variables: [49.]
Reward: -1.0
State #21
Game variables: [49.]
Reward: -1.0
State #22
Game variables: [49.]
Reward: -1.0
State #23
Game variables: [49.]
Reward: -1.0
State #24
Game variables: [49.]
Reward: -1.0
State #25
Game variables: [49.]
Reward: -1.0
State #26
Game variables: [49.]
Reward: -1.0
State #27
Game variables: [49.]
Reward: -1.0
State #28
Game variables: [49.]
Reward: -1.0
State #29
Game variables: [49.]
Reward: -6.0
State #30
Game variables: [48.]
Reward: -1.0
State #31
Game variables: [48.]
Reward: -1.0
State #32
Game variables: [48.]
Reward: -1.0
State #33
Game variables: [48.]
Reward: -1.0
State #34
Game variables: [48.]
Reward: -1.0
State #35
Game variables: [48.]
Reward: -1.0
State #36
Game variables: [48.]
Reward: -1.0
State #37
Game variables: [48.]
Reward: -1.0
State #38
Game variables: [48.]
Reward: -1.0
State #39
Game variables: [48.]
Reward: -1.0
State #40
Game variables: [48.]
Reward: -1.0
State #41
Game variables: [48.]
Reward: -1.0
State #42


State #146
Game variables: [42.]
Reward: -1.0
State #147
Game variables: [42.]
Reward: -1.0
State #148
Game variables: [42.]
Reward: -1.0
State #149
Game variables: [42.]
Reward: -1.0
State #150
Game variables: [42.]
Reward: -1.0
State #151
Game variables: [42.]
Reward: -1.0
State #152
Game variables: [42.]
Reward: -1.0
State #153
Game variables: [42.]
Reward: -6.0
State #154
Game variables: [41.]
Reward: -1.0
State #155
Game variables: [41.]
Reward: -1.0
State #156
Game variables: [41.]
Reward: -1.0
State #157
Game variables: [41.]
Reward: -1.0
State #158
Game variables: [41.]
Reward: -1.0
State #159
Game variables: [41.]
Reward: -1.0
State #160
Game variables: [41.]
Reward: -1.0
State #161
Game variables: [41.]
Reward: -1.0
State #162
Game variables: [41.]
Reward: -1.0
State #163
Game variables: [41.]
Reward: -1.0
State #164
Game variables: [41.]
Reward: -1.0
State #165
Game variables: [41.]
Reward: -1.0
State #166
Game variables: [41.]
Reward: -1.0
State #167
Game variables: [41.]
R

State #270
Game variables: [35.]
Reward: -1.0
State #271
Game variables: [35.]
Reward: -1.0
State #272
Game variables: [35.]
Reward: -1.0
State #273
Game variables: [35.]
Reward: -1.0
State #274
Game variables: [35.]
Reward: -1.0
State #275
Game variables: [35.]
Reward: -1.0
State #276
Game variables: [35.]
Reward: -1.0
State #277
Game variables: [35.]
Reward: -1.0
State #278
Game variables: [35.]
Reward: -1.0
State #279
Game variables: [35.]
Reward: -1.0
State #280
Game variables: [35.]
Reward: -1.0
State #281
Game variables: [35.]
Reward: -1.0
State #282
Game variables: [35.]
Reward: -1.0
State #283
Game variables: [35.]
Reward: -1.0
State #284
Game variables: [35.]
Reward: -1.0
State #285
Game variables: [35.]
Reward: -1.0
State #286
Game variables: [35.]
Reward: -6.0
State #287
Game variables: [34.]
Reward: -1.0
State #288
Game variables: [34.]
Reward: -1.0
State #289
Game variables: [34.]
Reward: -1.0
State #290
Game variables: [34.]
Reward: -1.0
State #291
Game variables: [34.]
R

State #29
Game variables: [48.]
Reward: -1.0
State #30
Game variables: [48.]
Reward: -1.0
State #31
Game variables: [48.]
Reward: -1.0
State #32
Game variables: [48.]
Reward: -1.0
State #33
Game variables: [48.]
Reward: -1.0
State #34
Game variables: [48.]
Reward: -1.0
State #35
Game variables: [48.]
Reward: -1.0
State #36
Game variables: [48.]
Reward: -1.0
State #37
Game variables: [48.]
Reward: -1.0
State #38
Game variables: [48.]
Reward: -1.0
State #39
Game variables: [48.]
Reward: -1.0
State #40
Game variables: [48.]
Reward: -6.0
State #41
Game variables: [47.]
Reward: -1.0
State #42
Game variables: [47.]
Reward: -1.0
State #43
Game variables: [47.]
Reward: -1.0
State #44
Game variables: [47.]
Reward: -1.0
State #45
Game variables: [47.]
Reward: -1.0
State #46
Game variables: [47.]
Reward: -1.0
State #47
Game variables: [47.]
Reward: -1.0
State #48
Game variables: [47.]
Reward: -1.0
State #49
Game variables: [47.]
Reward: -1.0
State #50
Game variables: [47.]
Reward: -1.0
State #51


State #155
Game variables: [41.]
Reward: -1.0
State #156
Game variables: [41.]
Reward: -1.0
State #157
Game variables: [41.]
Reward: -1.0
State #158
Game variables: [41.]
Reward: -1.0
State #159
Game variables: [41.]
Reward: -1.0
State #160
Game variables: [41.]
Reward: -1.0
State #161
Game variables: [41.]
Reward: -6.0
State #162
Game variables: [40.]
Reward: -1.0
State #163
Game variables: [40.]
Reward: -1.0
State #164
Game variables: [40.]
Reward: -1.0
State #165
Game variables: [40.]
Reward: -1.0
State #166
Game variables: [40.]
Reward: -1.0
State #167
Game variables: [40.]
Reward: -1.0
State #168
Game variables: [40.]
Reward: -1.0
State #169
Game variables: [40.]
Reward: -1.0
State #170
Game variables: [40.]
Reward: -1.0
State #171
Game variables: [40.]
Reward: -1.0
State #172
Game variables: [40.]
Reward: -1.0
State #173
Game variables: [40.]
Reward: -1.0
State #174
Game variables: [40.]
Reward: -1.0
State #175
Game variables: [40.]
Reward: -1.0
State #176
Game variables: [40.]
R

State #276
Game variables: [34.]
Reward: -1.0
State #277
Game variables: [34.]
Reward: -1.0
State #278
Game variables: [34.]
Reward: -1.0
State #279
Game variables: [34.]
Reward: -1.0
State #280
Game variables: [34.]
Reward: -1.0
State #281
Game variables: [34.]
Reward: -1.0
State #282
Game variables: [34.]
Reward: -1.0
State #283
Game variables: [34.]
Reward: -1.0
State #284
Game variables: [34.]
Reward: -1.0
State #285
Game variables: [34.]
Reward: -1.0
State #286
Game variables: [34.]
Reward: -1.0
State #287
Game variables: [34.]
Reward: -1.0
State #288
Game variables: [34.]
Reward: -1.0
State #289
Game variables: [34.]
Reward: -1.0
State #290
Game variables: [34.]
Reward: -1.0
State #291
Game variables: [34.]
Reward: -1.0
State #292
Game variables: [34.]
Reward: -1.0
State #293
Game variables: [34.]
Reward: -1.0
State #294
Game variables: [34.]
Reward: -6.0
State #295
Game variables: [33.]
Reward: -1.0
State #296
Game variables: [33.]
Reward: -1.0
State #297
Game variables: [33.]
R

State #94
Game variables: [45.]
Reward: -1.0
State #95
Game variables: [45.]
Reward: -1.0
State #96
Game variables: [45.]
Reward: -1.0
State #97
Game variables: [45.]
Reward: -1.0
State #98
Game variables: [45.]
Reward: -1.0
State #99
Game variables: [45.]
Reward: -1.0
State #100
Game variables: [45.]
Reward: -1.0
State #101
Game variables: [45.]
Reward: -1.0
State #102
Game variables: [45.]
Reward: -1.0
State #103
Game variables: [45.]
Reward: -1.0
State #104
Game variables: [45.]
Reward: -1.0
State #105
Game variables: [45.]
Reward: -1.0
State #106
Game variables: [45.]
Reward: -1.0
State #107
Game variables: [45.]
Reward: -1.0
State #108
Game variables: [45.]
Reward: -6.0
State #109
Game variables: [44.]
Reward: -1.0
State #110
Game variables: [44.]
Reward: -1.0
State #111
Game variables: [44.]
Reward: -1.0
State #112
Game variables: [44.]
Reward: -1.0
State #113
Game variables: [44.]
Reward: -1.0
State #114
Game variables: [44.]
Reward: -1.0
State #115
Game variables: [44.]
Reward:

State #217
Game variables: [39.]
Reward: -1.0
State #218
Game variables: [39.]
Reward: -1.0
State #219
Game variables: [39.]
Reward: -1.0
State #220
Game variables: [39.]
Reward: -1.0
State #221
Game variables: [39.]
Reward: -1.0
State #222
Game variables: [39.]
Reward: -1.0
State #223
Game variables: [39.]
Reward: -1.0
State #224
Game variables: [39.]
Reward: -1.0
State #225
Game variables: [39.]
Reward: -1.0
State #226
Game variables: [39.]
Reward: -1.0
State #227
Game variables: [39.]
Reward: -6.0
State #228
Game variables: [38.]
Reward: -1.0
State #229
Game variables: [38.]
Reward: -1.0
State #230
Game variables: [38.]
Reward: -1.0
State #231
Game variables: [38.]
Reward: -1.0
State #232
Game variables: [38.]
Reward: -1.0
State #233
Game variables: [38.]
Reward: -1.0
State #234
Game variables: [38.]
Reward: -1.0
State #235
Game variables: [38.]
Reward: -1.0
State #236
Game variables: [38.]
Reward: -1.0
State #237
Game variables: [38.]
Reward: -1.0
State #238
Game variables: [38.]
R

State #42
Game variables: [47.]
Reward: -1.0
State #43
Game variables: [47.]
Reward: -1.0
State #44
Game variables: [47.]
Reward: -1.0
State #45
Game variables: [47.]
Reward: -1.0
State #46
Game variables: [47.]
Reward: -1.0
State #47
Game variables: [47.]
Reward: -1.0
State #48
Game variables: [47.]
Reward: -1.0
State #49
Game variables: [47.]
Reward: -1.0
State #50
Game variables: [47.]
Reward: -1.0
State #51
Game variables: [47.]
Reward: -1.0
State #52
Game variables: [47.]
Reward: -1.0
State #53
Game variables: [47.]
Reward: -6.0
State #54
Game variables: [46.]
Reward: -1.0
State #55
Game variables: [46.]
Reward: -1.0
State #56
Game variables: [46.]
Reward: -1.0
State #57
Game variables: [46.]
Reward: -1.0
State #58
Game variables: [46.]
Reward: -1.0
State #59
Game variables: [46.]
Reward: -1.0
State #60
Game variables: [46.]
Reward: -1.0
State #61
Game variables: [46.]
Reward: -1.0
State #62
Game variables: [46.]
Reward: -1.0
State #63
Game variables: [46.]
Reward: -1.0
State #64


State #167
Game variables: [41.]
Reward: -6.0
State #168
Game variables: [40.]
Reward: -1.0
State #169
Game variables: [40.]
Reward: -1.0
State #170
Game variables: [40.]
Reward: -1.0
State #171
Game variables: [40.]
Reward: -1.0
State #172
Game variables: [40.]
Reward: -1.0
State #173
Game variables: [40.]
Reward: -1.0
State #174
Game variables: [40.]
Reward: -1.0
State #175
Game variables: [40.]
Reward: -1.0
State #176
Game variables: [40.]
Reward: -1.0
State #177
Game variables: [40.]
Reward: -1.0
State #178
Game variables: [40.]
Reward: -1.0
State #179
Game variables: [40.]
Reward: -1.0
State #180
Game variables: [40.]
Reward: -1.0
State #181
Game variables: [40.]
Reward: -1.0
State #182
Game variables: [40.]
Reward: -1.0
State #183
Game variables: [40.]
Reward: -1.0
State #184
Game variables: [40.]
Reward: -1.0
State #185
Game variables: [40.]
Reward: -1.0
State #186
Game variables: [40.]
Reward: -6.0
State #187
Game variables: [39.]
Reward: -1.0
State #188
Game variables: [39.]
R

State #292
Game variables: [34.]
Reward: -1.0
State #293
Game variables: [34.]
Reward: -1.0
State #294
Game variables: [34.]
Reward: -1.0
State #295
Game variables: [34.]
Reward: -1.0
State #296
Game variables: [34.]
Reward: -1.0
State #297
Game variables: [34.]
Reward: -1.0
State #298
Game variables: [34.]
Reward: -1.0
State #299
Game variables: [34.]
Reward: -1.0
State #300
Game variables: [34.]
Reward: -1.0
Episode finished.
Total reward: -380.0
************************
Episode #43
State #1
Game variables: [50.]
Reward: -1.0
State #2
Game variables: [50.]
Reward: -1.0
State #3
Game variables: [50.]
Reward: -1.0
State #4
Game variables: [50.]
Reward: -1.0
State #5
Game variables: [50.]
Reward: -6.0
State #6
Game variables: [49.]
Reward: -1.0
State #7
Game variables: [49.]
Reward: -1.0
State #8
Game variables: [49.]
Reward: -1.0
State #9
Game variables: [49.]
Reward: -1.0
State #10
Game variables: [49.]
Reward: -1.0
State #11
Game variables: [49.]
Reward: -1.0
State #12
Game variables

State #113
Game variables: [44.]
Reward: -1.0
State #114
Game variables: [44.]
Reward: -1.0
State #115
Game variables: [44.]
Reward: -1.0
State #116
Game variables: [44.]
Reward: -1.0
State #117
Game variables: [44.]
Reward: -1.0
State #118
Game variables: [44.]
Reward: -1.0
State #119
Game variables: [44.]
Reward: -1.0
State #120
Game variables: [44.]
Reward: -1.0
State #121
Game variables: [44.]
Reward: -1.0
State #122
Game variables: [44.]
Reward: -1.0
State #123
Game variables: [44.]
Reward: -1.0
State #124
Game variables: [44.]
Reward: -1.0
State #125
Game variables: [44.]
Reward: -1.0
State #126
Game variables: [44.]
Reward: -6.0
State #127
Game variables: [43.]
Reward: -1.0
State #128
Game variables: [43.]
Reward: -1.0
State #129
Game variables: [43.]
Reward: -1.0
State #130
Game variables: [43.]
Reward: -1.0
State #131
Game variables: [43.]
Reward: -1.0
State #132
Game variables: [43.]
Reward: -1.0
State #133
Game variables: [43.]
Reward: -1.0
State #134
Game variables: [43.]
R

State #239
Game variables: [38.]
Reward: -1.0
State #240
Game variables: [38.]
Reward: -1.0
State #241
Game variables: [38.]
Reward: -1.0
State #242
Game variables: [38.]
Reward: -1.0
State #243
Game variables: [38.]
Reward: -6.0
State #244
Game variables: [37.]
Reward: -1.0
State #245
Game variables: [37.]
Reward: -1.0
State #246
Game variables: [37.]
Reward: -1.0
State #247
Game variables: [37.]
Reward: -1.0
State #248
Game variables: [37.]
Reward: -1.0
State #249
Game variables: [37.]
Reward: -1.0
State #250
Game variables: [37.]
Reward: -1.0
State #251
Game variables: [37.]
Reward: -1.0
State #252
Game variables: [37.]
Reward: -1.0
State #253
Game variables: [37.]
Reward: -1.0
State #254
Game variables: [37.]
Reward: -1.0
State #255
Game variables: [37.]
Reward: -1.0
State #256
Game variables: [37.]
Reward: -1.0
State #257
Game variables: [37.]
Reward: -1.0
State #258
Game variables: [37.]
Reward: -1.0
State #259
Game variables: [37.]
Reward: -1.0
State #260
Game variables: [37.]
R

State #52
Game variables: [47.]
Reward: -1.0
State #53
Game variables: [47.]
Reward: -1.0
State #54
Game variables: [47.]
Reward: -1.0
State #55
Game variables: [47.]
Reward: -1.0
State #56
Game variables: [47.]
Reward: -1.0
State #57
Game variables: [47.]
Reward: -1.0
State #58
Game variables: [47.]
Reward: -6.0
State #59
Game variables: [46.]
Reward: -1.0
State #60
Game variables: [46.]
Reward: -1.0
State #61
Game variables: [46.]
Reward: -1.0
State #62
Game variables: [46.]
Reward: -1.0
State #63
Game variables: [46.]
Reward: -1.0
State #64
Game variables: [46.]
Reward: -1.0
State #65
Game variables: [46.]
Reward: -1.0
State #66
Game variables: [46.]
Reward: -1.0
State #67
Game variables: [46.]
Reward: -1.0
State #68
Game variables: [46.]
Reward: -1.0
State #69
Game variables: [46.]
Reward: -1.0
State #70
Game variables: [46.]
Reward: -1.0
State #71
Game variables: [46.]
Reward: -1.0
State #72
Game variables: [46.]
Reward: -1.0
State #73
Game variables: [46.]
Reward: -1.0
State #74


State #178
Game variables: [40.]
Reward: -1.0
State #179
Game variables: [40.]
Reward: -1.0
State #180
Game variables: [40.]
Reward: -1.0
State #181
Game variables: [40.]
Reward: -1.0
State #182
Game variables: [40.]
Reward: -1.0
State #183
Game variables: [40.]
Reward: -1.0
State #184
Game variables: [40.]
Reward: -1.0
State #185
Game variables: [40.]
Reward: -6.0
State #186
Game variables: [39.]
Reward: -1.0
State #187
Game variables: [39.]
Reward: -1.0
State #188
Game variables: [39.]
Reward: -1.0
State #189
Game variables: [39.]
Reward: -1.0
State #190
Game variables: [39.]
Reward: -1.0
State #191
Game variables: [39.]
Reward: -1.0
State #192
Game variables: [39.]
Reward: -1.0
State #193
Game variables: [39.]
Reward: -1.0
State #194
Game variables: [39.]
Reward: -1.0
State #195
Game variables: [39.]
Reward: -1.0
State #196
Game variables: [39.]
Reward: -1.0
State #197
Game variables: [39.]
Reward: -1.0
State #198
Game variables: [39.]
Reward: -1.0
State #199
Game variables: [39.]
R

State #2
Game variables: [50.]
Reward: -1.0
State #3
Game variables: [50.]
Reward: -1.0
State #4
Game variables: [50.]
Reward: -1.0
State #5
Game variables: [50.]
Reward: -1.0
State #6
Game variables: [50.]
Reward: -6.0
State #7
Game variables: [49.]
Reward: -1.0
State #8
Game variables: [49.]
Reward: -1.0
State #9
Game variables: [49.]
Reward: -1.0
State #10
Game variables: [49.]
Reward: -1.0
State #11
Game variables: [49.]
Reward: -1.0
State #12
Game variables: [49.]
Reward: -1.0
State #13
Game variables: [49.]
Reward: -1.0
State #14
Game variables: [49.]
Reward: -1.0
State #15
Game variables: [49.]
Reward: -1.0
State #16
Game variables: [49.]
Reward: -1.0
State #17
Game variables: [49.]
Reward: -1.0
State #18
Game variables: [49.]
Reward: -1.0
State #19
Game variables: [49.]
Reward: -1.0
State #20
Game variables: [49.]
Reward: -6.0
State #21
Game variables: [48.]
Reward: -1.0
State #22
Game variables: [48.]
Reward: -1.0
State #23
Game variables: [48.]
Reward: -1.0
State #24
Game var

State #12
Game variables: [49.]
Reward: -1.0
State #13
Game variables: [49.]
Reward: -1.0
State #14
Game variables: [49.]
Reward: -1.0
State #15
Game variables: [49.]
Reward: -1.0
State #16
Game variables: [49.]
Reward: -1.0
State #17
Game variables: [49.]
Reward: -1.0
State #18
Game variables: [49.]
Reward: -1.0
State #19
Game variables: [49.]
Reward: -1.0
State #20
Game variables: [49.]
Reward: -1.0
State #21
Game variables: [49.]
Reward: -1.0
State #22
Game variables: [49.]
Reward: -1.0
State #23
Game variables: [49.]
Reward: -1.0
State #24
Game variables: [49.]
Reward: -1.0
State #25
Game variables: [49.]
Reward: -6.0
State #26
Game variables: [48.]
Reward: -1.0
State #27
Game variables: [48.]
Reward: -1.0
State #28
Game variables: [48.]
Reward: -1.0
State #29
Game variables: [48.]
Reward: -1.0
State #30
Game variables: [48.]
Reward: -1.0
State #31
Game variables: [48.]
Reward: -1.0
State #32
Game variables: [48.]
Reward: -1.0
State #33
Game variables: [48.]
Reward: -1.0
State #34


State #135
Game variables: [43.]
Reward: -6.0
State #136
Game variables: [42.]
Reward: -1.0
State #137
Game variables: [42.]
Reward: -1.0
State #138
Game variables: [42.]
Reward: -1.0
State #139
Game variables: [42.]
Reward: -1.0
State #140
Game variables: [42.]
Reward: -1.0
State #141
Game variables: [42.]
Reward: -1.0
State #142
Game variables: [42.]
Reward: -1.0
State #143
Game variables: [42.]
Reward: -1.0
State #144
Game variables: [42.]
Reward: -1.0
State #145
Game variables: [42.]
Reward: -1.0
State #146
Game variables: [42.]
Reward: -1.0
State #147
Game variables: [42.]
Reward: -1.0
State #148
Game variables: [42.]
Reward: -1.0
State #149
Game variables: [42.]
Reward: -1.0
State #150
Game variables: [42.]
Reward: -1.0
State #151
Game variables: [42.]
Reward: -1.0
State #152
Game variables: [42.]
Reward: -1.0
State #153
Game variables: [42.]
Reward: -1.0
State #154
Game variables: [42.]
Reward: -1.0
State #155
Game variables: [42.]
Reward: -1.0
State #156
Game variables: [42.]
R

State #257
Game variables: [36.]
Reward: -1.0
State #258
Game variables: [36.]
Reward: -1.0
State #259
Game variables: [36.]
Reward: -1.0
State #260
Game variables: [36.]
Reward: -1.0
State #261
Game variables: [36.]
Reward: -1.0
State #262
Game variables: [36.]
Reward: -1.0
State #263
Game variables: [36.]
Reward: -1.0
State #264
Game variables: [36.]
Reward: -1.0
State #265
Game variables: [36.]
Reward: -1.0
State #266
Game variables: [36.]
Reward: -1.0
State #267
Game variables: [36.]
Reward: -1.0
State #268
Game variables: [36.]
Reward: -1.0
State #269
Game variables: [36.]
Reward: -1.0
State #270
Game variables: [36.]
Reward: -1.0
State #271
Game variables: [36.]
Reward: -1.0
State #272
Game variables: [36.]
Reward: -1.0
State #273
Game variables: [36.]
Reward: -1.0
State #274
Game variables: [36.]
Reward: -1.0
State #275
Game variables: [36.]
Reward: -6.0
State #276
Game variables: [35.]
Reward: -1.0
State #277
Game variables: [35.]
Reward: -1.0
State #278
Game variables: [35.]
R

State #78
Game variables: [46.]
Reward: -1.0
State #79
Game variables: [46.]
Reward: -6.0
State #80
Game variables: [45.]
Reward: -1.0
State #81
Game variables: [45.]
Reward: -1.0
State #82
Game variables: [45.]
Reward: -1.0
State #83
Game variables: [45.]
Reward: -1.0
State #84
Game variables: [45.]
Reward: -1.0
State #85
Game variables: [45.]
Reward: -1.0
State #86
Game variables: [45.]
Reward: -1.0
State #87
Game variables: [45.]
Reward: -1.0
State #88
Game variables: [45.]
Reward: -1.0
State #89
Game variables: [45.]
Reward: -1.0
State #90
Game variables: [45.]
Reward: -1.0
State #91
Game variables: [45.]
Reward: -1.0
State #92
Game variables: [45.]
Reward: -1.0
State #93
Game variables: [45.]
Reward: -1.0
State #94
Game variables: [45.]
Reward: -1.0
State #95
Game variables: [45.]
Reward: -1.0
State #96
Game variables: [45.]
Reward: -1.0
State #97
Game variables: [45.]
Reward: -1.0
State #98
Game variables: [45.]
Reward: -6.0
State #99
Game variables: [44.]
Reward: -1.0
State #100

State #202
Game variables: [40.]
Reward: -6.0
State #203
Game variables: [39.]
Reward: -1.0
State #204
Game variables: [39.]
Reward: -1.0
State #205
Game variables: [39.]
Reward: -1.0
State #206
Game variables: [39.]
Reward: -1.0
State #207
Game variables: [39.]
Reward: -1.0
State #208
Game variables: [39.]
Reward: -1.0
State #209
Game variables: [39.]
Reward: -1.0
State #210
Game variables: [39.]
Reward: -1.0
State #211
Game variables: [39.]
Reward: -1.0
State #212
Game variables: [39.]
Reward: -1.0
State #213
Game variables: [39.]
Reward: -1.0
State #214
Game variables: [39.]
Reward: -1.0
State #215
Game variables: [39.]
Reward: -1.0
State #216
Game variables: [39.]
Reward: -6.0
State #217
Game variables: [38.]
Reward: -1.0
State #218
Game variables: [38.]
Reward: -1.0
State #219
Game variables: [38.]
Reward: -1.0
State #220
Game variables: [38.]
Reward: -1.0
State #221
Game variables: [38.]
Reward: -1.0
State #222
Game variables: [38.]
Reward: -1.0
State #223
Game variables: [38.]
R

State #11
Game variables: [49.]
Reward: -1.0
State #12
Game variables: [49.]
Reward: -1.0
State #13
Game variables: [49.]
Reward: -1.0
State #14
Game variables: [49.]
Reward: -1.0
State #15
Game variables: [49.]
Reward: -1.0
State #16
Game variables: [49.]
Reward: -1.0
State #17
Game variables: [49.]
Reward: -1.0
State #18
Game variables: [49.]
Reward: -1.0
State #19
Game variables: [49.]
Reward: -1.0
State #20
Game variables: [49.]
Reward: -1.0
State #21
Game variables: [49.]
Reward: -1.0
State #22
Game variables: [49.]
Reward: -1.0
State #23
Game variables: [49.]
Reward: -1.0
State #24
Game variables: [49.]
Reward: -1.0
State #25
Game variables: [49.]
Reward: -1.0
State #26
Game variables: [49.]
Reward: -1.0
State #27
Game variables: [49.]
Reward: -1.0
State #28
Game variables: [49.]
Reward: -6.0
State #29
Game variables: [48.]
Reward: -1.0
State #30
Game variables: [48.]
Reward: -1.0
State #31
Game variables: [48.]
Reward: -1.0
State #32
Game variables: [48.]
Reward: -1.0
State #33


State #137
Game variables: [43.]
Reward: -1.0
State #138
Game variables: [43.]
Reward: -1.0
State #139
Game variables: [43.]
Reward: -6.0
State #140
Game variables: [42.]
Reward: -1.0
State #141
Game variables: [42.]
Reward: -1.0
State #142
Game variables: [42.]
Reward: -1.0
State #143
Game variables: [42.]
Reward: -1.0
State #144
Game variables: [42.]
Reward: -1.0
State #145
Game variables: [42.]
Reward: -1.0
State #146
Game variables: [42.]
Reward: -1.0
State #147
Game variables: [42.]
Reward: -1.0
State #148
Game variables: [42.]
Reward: -1.0
State #149
Game variables: [42.]
Reward: -1.0
State #150
Game variables: [42.]
Reward: -1.0
State #151
Game variables: [42.]
Reward: -1.0
State #152
Game variables: [42.]
Reward: -1.0
State #153
Game variables: [42.]
Reward: -6.0
State #154
Game variables: [41.]
Reward: -1.0
State #155
Game variables: [41.]
Reward: -1.0
State #156
Game variables: [41.]
Reward: -1.0
State #157
Game variables: [41.]
Reward: -1.0
State #158
Game variables: [41.]
R

Game variables: [36.]
Reward: -6.0
State #259
Game variables: [35.]
Reward: -1.0
State #260
Game variables: [35.]
Reward: -1.0
State #261
Game variables: [35.]
Reward: -1.0
State #262
Game variables: [35.]
Reward: -1.0
State #263
Game variables: [35.]
Reward: -1.0
State #264
Game variables: [35.]
Reward: -1.0
State #265
Game variables: [35.]
Reward: -1.0
State #266
Game variables: [35.]
Reward: -1.0
State #267
Game variables: [35.]
Reward: -1.0
State #268
Game variables: [35.]
Reward: -1.0
State #269
Game variables: [35.]
Reward: -1.0
State #270
Game variables: [35.]
Reward: -1.0
State #271
Game variables: [35.]
Reward: -1.0
State #272
Game variables: [35.]
Reward: -1.0
State #273
Game variables: [35.]
Reward: -1.0
State #274
Game variables: [35.]
Reward: -1.0
State #275
Game variables: [35.]
Reward: -1.0
State #276
Game variables: [35.]
Reward: -1.0
State #277
Game variables: [35.]
Reward: -6.0
State #278
Game variables: [34.]
Reward: -1.0
State #279
Game variables: [34.]
Reward: -1.0

Benchmarks (means calculated over 50 runs)
    - scenario "Basic":                         benchmark_Basic = -160
    - scenario "Health gathering supreme":      benchmark_HGS = +300
    - scenario "Deadly corridor":               benchmark_DC = -115
    